controller/main.js

```diff
@@ -193,7 +193,11 @@ _.each(view.list, function (elem) {
   //elem.unkey(['C-d']);
 
   elem.key(['C-d'], function () {
-    main.show(diff);
+    var canDiff = view.list.unstaged.interactive ? git.files.unstaged.length : git.files.staged.length;
+
+    if (canDiff) {
+      main.show(diff);
+    }
   });
 
```

Comment:
I think, it's better the checking codes will be in **controller/diff.js**

Add below codes before the line 8 in the diff.js;

```js
if (parent.prevFocused.selected === undefined) {
  return;
}
```

It's cleaner and makes easy to refactor. :-) (something like adding error messages box)